app/controllers/conversations_controller.rb

```diff
@@ -0,0 +1,54 @@
+class ConversationsController < ApplicationController
+  before_filter :authenticate_user!, :mailbox
+  before_filter :conversation, except: [:index, :new, :create]
+
+  def create
+    recipient_ids = conversation_params[:recipients].split(',')
+    recipients = User.where(id: recipient_ids).all
+
+    conversation = current_user.
+      send_message(recipients, conversation_params[:body], conversation_params[:subject]).conversation
+
+    redirect_to conversation_path(conversation)
+  end
+
+  def show
+
+    # mark as read
+    current_user.mark_as_read @conversation
+  end
+
+  def reply
+    current_user.reply_to_conversation(conversation, message_params[:body], message_params[:subject])
+    redirect_to conversation_path(conversation)
+  end
+
+  def trash
+    conversation.move_to_trash(current_user)
+    redirect_to :conversations
+  end
+
+  def untrash
+    conversation.untrash(current_user)
+    redirect_to :conversations
+  end
+
+  private
+
+  def mailbox
+    @mailbox ||= current_user.mailbox
+  end
+
+  def conversation
+    @conversation ||= @mailbox.conversations.find(params[:id])
+  end
+
+  def conversation_params
+    params.require(:conversation).permit(:recipients, :subject, :body)
```

Comment:
If `:recipients` is an array, strong params can help:

```ruby
params.require(:conversation).permit({recipients: []}, :subject, :body)
```

Review this section and sub sections for examples of the advanced data structures strong params can handle: http://guides.rubyonrails.org/action_controller_overview.html#strong-parameters